### DISCUSSION TITLE
Disable some plugins based on current slide

### DISCUSSION THREAD

**Discussion by madalinul** [igordanchenko/yet-another-react-lightbox#177] [Q&A] [answered]
Is there a way to disable a plugin based on the current slide? For example I have some slides with images and some slides with custom content and I would like to disable the zoom plugin when the current slide is not an image.

**Comment by igordanchenko** [comment, marked as answer]
Zoom plugin buttons are automatically disabled on unsupported slides - that's the UX approach implemented across all plugins. Do you want to completely hide zoom buttons on unsupported slides? Please note that this would cause toolbar buttons layout shift while navigating between slides - which is probably not the best UX.

Here is how you can conditionally render plugins based on the current slide:

```jsx
const [open, setOpen] = React.useState(false);
const [index, setIndex] = React.useState(0);

// ...

<Lightbox
    open={open}
    close={() => setOpen(false)}
    index={index}
    slides={slides}
    plugins={index !== 1 ? [Counter] : []}
    on={{ view: ({ index: currentIndex }) => setIndex(currentIndex) }}
/>
```

However, conditional rendering of some plugins may interfere with slides' swipe animation, and that's exactly what's happening in the case of Zoom plugin.

An alternative approach would be to hide zoom buttons with css:

```jsx
const [open, setOpen] = React.useState(false);
const [index, setIndex] = React.useState(0);

// ...

<Lightbox
    open={open}
    close={() => setOpen(false)}
    slides={slides}
    index={index}
    plugins={[Zoom]}
    on={{ view: ({ index: currentIndex }) => setIndex(currentIndex) }}
    className={index === 1 ? "hide-zoom-buttons" : undefined}
/>
```

```css
.hide-zoom-buttons .yarl__button[aria-label="Zoom in"],
.hide-zoom-buttons .yarl__button[aria-label="Zoom out"] {
    display: none;
}
```
